packages/next/src/client/app-index.tsx

```diff
@@ -122,11 +135,14 @@ const DOMContentLoaded = function () {
   }
   initialServerDataLoaded = true
 }
+
+const queueTask = queueMicrotask || ((fn: Function) => setTimeout(fn, 0))
```

Comment:
Do you think a microTask is sufficient, or do you want to have the code to always run to completion first (macroTask)? Otherwise there is still the chance of race conditions from multiple nested microtasks

Reply:
Want to delay it in macro much later, moved to setTimeout (marco task)

Reply:
If we want it to be deferred after a macrotask, `setImmediate` will be faster than `setTimeout` (which has a minimal delay of 4ms in some cases).

Reply:
we only have setTimeout here since it's in client